rtrlib/pfx/pfx.h

```diff
@@ -175,5 +182,30 @@ void pfx_table_for_each_ipv4_record(struct pfx_table *pfx_table, pfx_for_each_fp
  */
 void pfx_table_for_each_ipv6_record(struct pfx_table *pfx_table, pfx_for_each_fp fp, void *data);
 
+/**
+ * @brief Copy content of src into dst
+ * @details dst must be empty and initialized
+ * @param[in] src_table Source table
+ * @param[in] dst_table Destination table
+ * @param[in] socket socket which prefixes should not be copied
+ */
+void pfx_table_copy_except_socket(struct pfx_table *src_table, struct pfx_table *dst_table, const struct rtr_socket *socket);
```

Comment:
src_table could be const, too?

Reply:
Unfortunately no, this function calls `pfx_table_for_each_ipv{4,6}_record` which must mutate the source table to acquire a read lock.